### PR TITLE
Align waves2amr initialization with other wave profiles; add +abl reg test

### PR DESCRIPTION
**Pull request by mbkuhn**
## Summary

Follow on to #1354 for the waves2amr pathway. Added a case with an ABL, inspired because this PR fixes flaws in the initialization when combined with other physics.

## Pull request type

Please check the type of change introduced:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [ ] new unit-test(s)
- [x] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [ ] on CPU 
- the regression tests
  - [ ] on GPU 
  - [ ] on CPU 

## Additional background

Issue Number:

### CODE REVIEW

**Comment by mbkuhn** [comment]
I am going to add some documentation changes to this PR too. @moprak-nrel this should be safe for existing workflows; I confirmed that the new and old w2a reg tests run, requesting a code quality check

**Comment by moprak-nrel** [comment]
Looks good to me! Just need the CI for the docs to pass.

File: amr-wind/ocean_waves/relaxation_zones/waves2amr_ops.H [hunk 1]
```diff
@@ -457,20 +439,61 @@ struct InitDataOp<W2AWaves>
...
-        auto& ow_velocity = sim.repo().get_field("ow_velocity");
-        amrex::MultiFab::Copy(ow_levelset(level), levelset(level), 0, 0, 1, 3);
-        amrex::MultiFab::Copy(
-            ow_velocity(level), velocity(level), 0, 0, AMREX_SPACEDIM, 3);
+        // Zero velocity in pure air cells
+        postprocess_velocity_mfab_liquid(
+            ow_velocity(level), ow_levelset(level), dx);
+
+        // Populate flow fields according to intended forcing and init setup
+        for (amrex::MFIter mfi(levelset(level)); mfi.isValid(); ++mfi) {
```
Comment (marchdf) [inline amr-wind/ocean_waves/relaxation_zones/waves2amr_ops.H:452]: Can we make this a fused MF now? That will make it work with openmp out of the box and better on GPU in general.

Comment (mbkuhn) [reply amr-wind/ocean_waves/relaxation_zones/waves2amr_ops.H:452]: yep, I should add that to this.

**Comment by mbkuhn** [comment]
replaced all the MFIters in ocean_waves/

**Review by moprak-nrel — APPROVED** [review]

**Pull request merged**